Nodes analyzer: v0.2 refatoring

codeliner opened the pull request:
This PR contains a complete rewrite of the internal message flow structure. I've simplified the structure to only use `nodes` and `edges` instead of messages, handlers, aggregates, ... Those are just node types now so it is easier to understand and customize the message flow. 
And the message flow can now directly be imported into the mgmt UI. This resolves #15 

Other output formatters are removed but you can still write your own if needed.

Documentation is also included in the PR.

Commit:
v0.2 Refactoring

Collect nodes and edges within the message flow

Commit:
Rm no longer supported output formatters

Commit:
Catch IdentifierNotFound errors

Commit:
Nodes and edges should be an array

Commit:
Use node name instead of title

Commit:
Replace pro icon with free one

Commit:
Introduce parentColor

Commit:
Use NodeIcon and expand options to also use fa brand/regular, links

Commit:
Check if event recorder calls others recording methods

Commit:
Check if invoked recorders are found

Commit:
CS fixes

Commit:
Add finalizer

Commit:
Configurable node class

Commit:
Configure travis.ci

Commit:
Add badges and support hints

Commit:
Add docs

Commit:
Add more docs

Commit:
Add link to prooph-do

kochen commented:
I just gave it a try and got:
```
PHP Fatal error:  Uncaught Error: Class 'MessageHandlerCollector' not found in .../vendor/prooph/message-flow-analyzer/src/Helper/ProjectTraverserFactory.php:69
Stack trace:
#0 .../vendor/prooph/message-flow-analyzer/src/Cli/AnalyzeProjectCommand.php(90): Prooph\MessageFlowAnalyzer\Helper\ProjectTraverserFactory::buildTraverserFromConfig(Array)
#1 .../vendor/symfony/console/Command/Command.php(252): Prooph\MessageFlowAnalyzer\Cli\AnalyzeProjectCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#2 .../vendor/symfony/console/Application.php(865): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#3 .../vendor/symfony/console/Application.php(241): Symfony\Component\Console\Application->doRunCommand(Objec in .../vendor/prooph/message-flow-analyzer/src/Helper/ProjectTraverserFactory.php on line 69
```
Same thing for `EventRecorderCollector`, `EventRecorderCollector`, `EventRecorderInvokerCollector`.


Could you add a link to a new "default" `prooph_analyzer.json` configuration file, based on this: https://github.com/prooph/message-flow-analyzer/blob/nodes_analyser/docs/configuration.md#classvisitors
?

```
{
  "name": "Default Project",
  "fileInfoFilters": [
    "ExcludeVendorDir",
    "ExcludeTestsDir",
    "ExcludeHiddenFileInfo",
    "IncludePHPFile"
  ],
  "classVisitors": [
    "MessageCollector",
    "CommandHandlerCollector",
    "MessageProducerCollector",
    "AggregateMethodCollector",
    "EventListenerCollector"
  ]
}
```

kochen commented:
@codeliner First run, finished OK, but I got only `Nodes` - no `Edges`.
Tried to run it again and got:
```
Edge with id e5fa07cd10ad6806e3098db16b3793724418a228_6f4a1d2478dd80d681f0e34da8cd07f2c0812106 is already set. Got {"data":{"id":"e5fa07cd10ad6806e3098db16b3793724418a228_6f4a1d2478dd80d681f0e34da8cd07f2c0812106","source":"e5fa07cd10ad6806e3098db16b3793724418a228","target":"6f4a1d2478dd80d681f0e34da8cd07f2c0812106"}}
```
even after clearing the cache and deleting the output file.

codeliner commented:
@kochen once the PR is merged the link to example config will reference the new example config file

codeliner commented:
@kochen regarding the problem with no edges. You run it against your own project right?
Any chance you can give me more information why no edges are added?
It runs perfectly fine against proophessor-do but that was my only testing playground so far.
Going to run it against one of our production systems but I alrready know that I need project specific class visitors for the project bc we work with some additions to the OSS version of prooph/event-sourcing in that project.

Commit:
Always use $messageFlow->setEdge

codeliner commented:
@kochen can you try again with my latest changes? I've removed `$messageFlow->addEdge()` and always use `$messageFlow->setEdge()` now, so duplicate edges should no longer cause exceptions.

Commit:
Add file info in case of an exception

Commit:
Set edge for message producer services

Commit:
Rm namespace from message producer service name

Commit:
Detect event recorders fetched from inherited methods

Commit:
Get and remove node/edge methods

Commit:
Remove linked edges of removed node

Commit:
Replace listener and producer collectors w/ messaging collector

Commit:
Configurable EventRecorder check function

Commit:
Fix wrong unset in removeNode

Commit:
Fix wrong unset in removeEdge

Commit:
Detect message production if factory is used

Commit:
Check if $node->var->name is set

Commit:
Remove method filter public for event listener checks

A public method meight only be used as proxy but internal
methods handle specific events. We're interested in this flow
so we should scan all methods

Commit:
Message producers can create more than one message

kochen commented:
@codeliner I just tried with the (current) latest changes and the result is the same - no edges between nodes.
I've tried limiting is to one `src/{context}` and not the entire `src`.

My domain classes are mostly based on those from the `Proophesor-do`.

codeliner commented:
@kochen I'm analyzing a very large project with this branch. It definitely works so we need to find out why it is not working for you.

Can you post some code snippets like a command handler, an aggregate, one or two commands, events?

Do you use default prooph classes or some specials?

Commit:
Allow symfony/console 3.2 as well

Commit:
All scan helper funcs should be public
